js/src/ui/TxList/store.js

```diff
@@ -14,150 +14,121 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-import { action, observable, transaction } from 'mobx';
-import { uniq } from 'lodash';
+import { action, observable } from 'mobx';
 
 export default class Store {
   @observable blocks = {};
   @observable sortedHashes = [];
   @observable transactions = {};
 
-  constructor (api) {
+  constructor (api, newError) {
     this._api = api;
-    this._subscriptionId = 0;
-    this._pendingHashes = [];
-
-    this.subscribe();
+    this._newError = newError;
   }
 
-  @action addBlocks = (blocks) => {
-    this.blocks = Object.assign({}, this.blocks, blocks);
+  @action addHash = (hash) => {
+    if (!this.sortedHashes.includes(hash)) {
+      this.sortedHashes.push(hash);
+      this.sortHashes();
+    }
   }
 
-  @action addTransactions = (transactions) => {
-    transaction(() => {
-      this.transactions = Object.assign({}, this.transactions, transactions);
-      this.sortedHashes = Object
-        .keys(this.transactions)
-        .sort((ahash, bhash) => {
-          const bnA = this.transactions[ahash].blockNumber;
-          const bnB = this.transactions[bhash].blockNumber;
-
-          if (bnB.eq(0)) {
-            return bnB.eq(bnA) ? 0 : 1;
-          } else if (bnA.eq(0)) {
-            return -1;
-          }
-
-          return bnB.comparedTo(bnA);
-        });
-
-      this._pendingHashes = this.sortedHashes.filter((hash) => this.transactions[hash].blockNumber.eq(0));
-    });
+  @action removeHash = (hashes) => {
```

Comment:
Ok, I think I get the context in which this function is used, but `removeHash` seems like a function that should do the invert of `addHash` (given hash add it to the list if not present) so it should do: "given hash remove it from the list if present". This function is actually just overriding all hashes with new ones that's why I though it should be named `setHashes` instead of `removeHash`.